### DISCUSSION TITLE
Qual a melhor maneira para criar páginas com estruturas diferentes no WordPress?

### DISCUSSION THREAD

**Discussion by altemberg** [frontendbr/forum#889] [Perguntas e Respostas]
Galera,

Eu não curtia muito o WordPress por conta de alguns trabalhos que acabei pegando pra dar manutenção e fiquei com uma imagem muito ruim dele! Mas aí esse ano resolvi dar uma chance e aprender como esse danado funciona, e pra minha surpresa, tudo o que tive foi uma péssima primeira impressão e agora tô curtindo pra caramba aprender como ele funciona.

Já venho estudando faz uns dias e agora estou partindo pra fazer as coisas sozinho, então peguei alguns PSD's e estou começando a passar eles pro WP.

Mas aí me bateu a seguinte dúvida: O template que estou montando tem umas 4 páginas com estruturas totalmente diferentes, então eu acho que não daria pra usar apenas um modelo de página pra elas, correto?

Andei pesquisando e vi que muita gente usa Page Builders para problemas assim, e eu particularmente não curto muito eles, as vezes que fiz alguma coisa com WordPress foram com temas que usavam Page Builder e eu achei muito ruim, pesado pra caramba e não queria ter que recorrer a algo assim.

A princípio eu tinha pensado em fazer alguns CPT's e fazer um modelo pra cada página. Pelo o que já vi, acho que realmente seja essa a solução, mas pra não ficar com essa dúvida preferi recorrer a quem entende mais :P

**Comment by renanlara** [comment]
Cara, é muito simples! Pode fazer isso de diversas formas: usando o ID da página (page-ID.php), com o nome da página (page-nome-da-pagina.php e usando [Templates Pages do WordPress](https://developer.wordpress.org/themes/template-files-section/page-template-files/).

Qualquer dúvida conte comigo!

**Comment by tadeubrasil** [comment]
Melhor coisa é ir na fonte!!
[https://developer.wordpress.org/themes/template-files-section/page-template-files/](https://developer.wordpress.org/themes/template-files-section/page-template-files/)

Tiver dúvidas estamos ai!!

**Comment by altemberg** [comment]
@renanlara @tadeubrasil Valeu, galera!
Outra coisa, seria uma prática ruim fazer o uso de vários templates?

**Comment by renanlara** [comment]
@altemberg, não. Isso não irá afetar em nada. Só precisa ter em mente a boa e velha otimização do carregamento das páginas.

**Comment by theandersonn** [comment]
@altemberg, acrescentando um detalhe sobre as páginas que você pode trabalhar com os slugs, que são lidos dinamicamente pelo WordPress, exemplo: page-contato.php o slug 'contato será lido dinamicamente, ou seja criar a página no admin e anexar esse template somente para exibi-la'.

Se tiver partes similares, poderá reutilizar o código com 'get template part'
https://codex.wordpress.org/pt-br:get_template_part

E sobre criar um novo CPT, tenha em mente que eles são novos fluxos de gerenciamento.

Dois boilerplates que montei e utilizo em projetos que desenvolvo são:

 **Temas**
https://github.com/theandersonn/blank-theme-wp

**CPTs**
https://github.com/theandersonn/custom-post-types

_E para funcionalidades marotas do dia a dia:_
https://github.com/theandersonn/wp-snippets

**Comment by altemberg** [comment]
@theandersonn cara, esses boilerplates vão me ajudar muito, inclusive, pra estudar, valeuzão!

**Comment by theandersonn** [comment]
Beleza, show bola! Qualquer coisa, se surgir dúvidas, poste aí :)

**Comment by luizbills** [comment]
O plugin [Timber](https://github.com/timber/timber) também pode te ajudar bastante.
